### ISSUE TITLE
Encrypt password to get the plugin?

### ISSUE THREAD

**Issue by mtessler-fmcc** [etiennestuder/gradle-credentials-plugin#11]
Has this been though through?
I am trying to figure out how to encrypt the property for the password for the repository where the plugin is stored?
How do I use the plugin to decrypt the password to retrieve the plugin?

```
buildscript {
  String nexusPasswd = credentials.nexusPasswd
  maven {
    url "https://xxx.myNexusRepo.nowhere"
    credentials {
        username = nexusUser
        password = nexusPasswd
    }
  }
  dependencies {
    classpath 'nu.studer:gradle-credentials-plugin:1.0.1'
  }
}

apply plugin: 'nu.studer.credentials'
```

**Comment by caseykulm** [comment]
I also am looking for this functionality, however I do not believe this is currently possible. I am personally working on a solution at the moment, and may update this comment if I make any progress.

Seems like since the Plugin is of type Plugin<Project> that it cannot be applied in the maven closure. A plugin could possibly be used in the maven closure if it was of type Plugin<MavenArtifactRepository>. 

There's also this class https://docs.gradle.org/current/dsl/org.gradle.plugin.repository.MavenPluginRepositoyry.html

Basically, just like jcenter() and mavenCentral() are custom MavenArtifactRepository object, we might be able to make a GradleSecureMavenArtifactRepository, or at least get there with a plugin

**Comment by sustacek** [comment]
I might be mistaken, but the original comment is asking about using the functionality (= retrieve password from credentials store) of the plugin to get the plugin (its .pom & .jar) itself, from a protected Nexus repository. Is that right?

If so, I'm afraid that's not possible. `project.credentials` is available in the code only after the plugin is applied (`apply plugin: 'nu.studer.credentials'`) on the project. And to apply the plugin, you have to fetch it, it's you don't have it locally already.

Why not get the plugin from `jcenter()`? Or the auth is for getting another plugin, not the 'credentials' plugin?

**Comment by etiennestuder** [comment]
This cannot be done. Closing.
